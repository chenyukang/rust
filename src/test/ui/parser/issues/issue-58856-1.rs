impl A {
    //~^ ERROR cannot find type `A` in this scope
    fn b(self>
    //~^ ERROR expected parameter name
    //~^^ ERROR expected one of
    //~| ERROR associated function in `impl` without body
}
//~^ ERROR mismatched closing
//~| ERROR expected one of

fn main() {}
