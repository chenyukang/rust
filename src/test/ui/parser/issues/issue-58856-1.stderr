error: mismatched closing delimiter: `}`
  --> $DIR/issue-58856-1.rs:6:1
   |
LL |     fn b(self>
   |         - unmatched opening delimiter begins here
...
LL | }
   | ^ mismatched closing delimiter

error: expected parameter name, found `>`
  --> $DIR/issue-58856-1.rs:3:14
   |
LL |     fn b(self>
   |              ^ expected parameter name

error: expected one of `)`, `,`, or `:`, found `>`
  --> $DIR/issue-58856-1.rs:3:14
   |
LL |     fn b(self>
   |              ^
   |              |
   |              expected one of `)`, `,`, or `:`
   |              help: missing `,`

error: expected one of `->`, `where`, or `{`, found `}`
  --> $DIR/issue-58856-1.rs:6:1
   |
LL |     fn b(self>
   |        - while parsing this `fn`
...
LL | }
   | ^ expected one of `->`, `where`, or `{`

error: associated function in `impl` without body
  --> $DIR/issue-58856-1.rs:3:5
   |
LL | /     fn b(self>
LL | |
LL | |
LL | | }
   | | ^- help: provide a definition for the function: `{ <body> }`
   | |_|
   | 

error[E0412]: cannot find type `A` in this scope
  --> $DIR/issue-58856-1.rs:1:6
   |
LL | impl A {
   |      ^ not found in this scope

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0412`.
